src/neorados/cls/log.cc

```diff
@@ -0,0 +1,132 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+/*
+ * Ceph - scalable distributed file system
+ *
+ * Copyright (C) 2023 IBM
+ *
+ * See file COPYING for license information.
+ *
+ */
+
+#include "log.h"
+
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "include/buffer.h"
+#include "common/ceph_time.h"
+
+#include "include/neorados/RADOS.hpp"
+
+
+#include "cls/log/cls_log_types.h"
+#include "cls/log/cls_log_ops.h"
+
+namespace neorados::cls::log {
+using boost::system::error_code;
+namespace buffer = ceph::buffer;
+
+void add(neorados::WriteOp& op, std::vector<entry> entries)
+{
+  buffer::list in;
+  ::cls::log::ops::add_op call;
+  call.entries = std::move(entries);
+  encode(call, in);
+  op.exec("log", "add", in);
+}
+
+void add(neorados::WriteOp& op, entry e)
+{
+  bufferlist in;
+  ::cls::log::ops::add_op call;
+  call.entries.push_back(std::move(e));
+  encode(call, in);
+  op.exec("log", "add", in);
+}
+
+void add(neorados::WriteOp& op, ceph::real_time timestamp,
+	 std::string section, std::string name,
+	 buffer::list&& bl)
+{
+  bufferlist in;
+  ::cls::log::ops::add_op call;
+  call.entries.emplace_back(timestamp, std::move(section),
+			    std::move(name), std::move(bl));
+  encode(call, in);
+  op.exec("log", "add", in);
+}
+
+void list(ReadOp& op, ceph::real_time from, ceph::real_time to,
+	  std::optional<std::string> in_marker, std::span<entry>& entries,
```

Comment:
i think we'd need to take the span by-value in order to take advantage of implicit conversions from array/vector?

Reply:
Fixed